packages/core-v2-graphql/src/utils/getDataSourceByAddress.ts

```diff
@@ -3,8 +3,9 @@ import { CoreV2Context } from "src/context";
 export function getDataSourceByAddress(
   address: string,
   dataSources: CoreV2Context["elementDataSources"],
+  key: keyof CoreV2Context["elementDataSources"],
 ): any | undefined {
-  return dataSources.find(
+  return dataSources[key].find(
     (dataSource: { address: string }) => dataSource.address === address,
   );
 }
```

Comment:
Oh this is cool.

What if we made it even simpler like this:

```typescript
type CoreV2DataSources = CoreV2Context["elementDataSources"]

export function getDataSourceByAddress(
  address: string,
  dataSources: CoreV2DataSources[keyof CoreV2DataSources],
): any | undefined {
  return dataSources.find(
    (dataSource: { address: string }) => dataSource.address === address,
  );
}
```

So then I could call it like this:

```typescript
const tokenContract = getDataSourceByAddress(
  address,
  context.elementDataSources.tokenContracts,
);
```

Reply:
dope! also added a helper type so we can get a proper return type instead of any